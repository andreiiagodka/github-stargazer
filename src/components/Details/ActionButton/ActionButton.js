import React from 'react'
import PropTypes from 'prop-types'

import { Button } from 'react-bootstrap'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTrash } from '@fortawesome/free-solid-svg-icons'

import { deleteRepository } from '../../../shared/firebase'

const ActionButton = ({ id, history }) => {
  const handleDelete = () => {
    deleteRepository(id)
    history.push('/')
  }

  return (
    <Button variant='light' onClick={handleDelete}>
      <FontAwesomeIcon icon={faTrash} />
    </Button>
  )
}

ActionButton.propTypes = {
  id: PropTypes.string.isRequired,
  history: PropTypes.object.isRequired
}

export default ActionButton