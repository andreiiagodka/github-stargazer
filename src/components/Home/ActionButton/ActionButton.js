import React from 'react'

import { Button } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus } from '@fortawesome/free-solid-svg-icons'

const ActionButton = () => (
  <Link to='/new'>
    <Button variant='light'>
      <FontAwesomeIcon icon={faPlus} />
    </Button>
  </Link>
)

export default ActionButton