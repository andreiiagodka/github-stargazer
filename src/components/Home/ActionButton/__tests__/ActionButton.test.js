import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import ActionButton from '../ActionButton'

describe('<ActionButton /> component', () => {
  it('renders component correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <ActionButton />
      </MemoryRouter>
    ).toJSON()
  
    expect(tree).toMatchSnapshot()
  })
})
