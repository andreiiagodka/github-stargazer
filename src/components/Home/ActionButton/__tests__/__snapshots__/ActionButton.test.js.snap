// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton /> component renders component correctly 1`] = `
<a
  href="/new"
  onClick={[Function]}
>
  <button
    className="btn btn-light"
    disabled={false}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-w-14 "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
</a>
`;
